<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1440</width>
    <height>900</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1024</width>
    <height>768</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1440</width>
    <height>900</height>
   </size>
  </property>
  <property name="font">
   <font>
    <pointsize>12</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>DelfosExecution</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="cmdOpen">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>10</y>
      <width>101</width>
      <height>101</height>
     </rect>
    </property>
    <property name="text">
     <string>Open File</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="txtPathFile">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>10</y>
      <width>500</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>/home/pi/battery/default/default.yml</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cmdStop">
    <property name="geometry">
     <rect>
      <x>1350</x>
      <y>10</y>
      <width>81</width>
      <height>81</height>
     </rect>
    </property>
    <property name="text">
     <string>Stop</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cmdRun">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>10</y>
      <width>91</width>
      <height>81</height>
     </rect>
    </property>
    <property name="text">
     <string>Run</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblPicture">
    <property name="geometry">
     <rect>
      <x>1</x>
      <y>0</y>
      <width>900</width>
      <height>900</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string>Picture</string>
    </property>
   </widget>
   <widget class="QListWidget" name="lstErrors">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>360</y>
      <width>501</width>
      <height>231</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="lblErrors">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>340</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Errors</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cmdLights">
    <property name="geometry">
     <rect>
      <x>1270</x>
      <y>600</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Turn On Lights</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="cmdConfigs">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>599</y>
      <width>100</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Configs</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="grpConfigs">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>650</y>
      <width>351</width>
      <height>221</height>
     </rect>
    </property>
    <property name="title">
     <string>Configs</string>
    </property>
    <widget class="QGroupBox" name="grpLights">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>161</width>
       <height>191</height>
      </rect>
     </property>
     <property name="title">
      <string>Lights</string>
     </property>
     <widget class="QLabel" name="lblLightUp">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>60</y>
        <width>100</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Light Up</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lblLightDown">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>90</y>
        <width>100</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Light Down</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lblLightLeft">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>120</y>
        <width>100</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Light Left</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lblLightRight">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>150</y>
        <width>100</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Light Right</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtLightUp">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>60</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtLightDown">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>90</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtLightLeft">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>120</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtLightRight">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>150</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLabel" name="lblLocation">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>20</y>
        <width>91</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Location</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lblOutput">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>20</y>
        <width>71</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Output</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="grpIOs">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>20</y>
       <width>161</width>
       <height>191</height>
      </rect>
     </property>
     <property name="title">
      <string>IOs</string>
     </property>
     <widget class="QLineEdit" name="txtSensorUnit">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>60</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLabel" name="lblSensorUnit">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>60</y>
        <width>100</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Sensor Unit</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtSolenoid">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>90</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLabel" name="lblSolenoid">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>90</y>
        <width>100</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Stopper</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lblSignal">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>20</y>
        <width>100</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Signal</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lblIOs">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>20</y>
        <width>51</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>I/Os</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtButton">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>120</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLabel" name="lblButton">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>120</y>
        <width>100</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Button</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtWindows">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>150</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLabel" name="lblWindows">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>150</y>
        <width>100</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Windows</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="lblRows">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>100</y>
      <width>47</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Rows</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblColumns">
    <property name="geometry">
     <rect>
      <x>1150</x>
      <y>100</y>
      <width>71</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Columns</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblNumberRows">
    <property name="geometry">
     <rect>
      <x>1060</x>
      <y>100</y>
      <width>61</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="lblNumberColumns">
    <property name="geometry">
     <rect>
      <x>1220</x>
      <y>100</y>
      <width>61</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="lblStatus">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>160</y>
      <width>501</width>
      <height>161</height>
     </rect>
    </property>
    <property name="palette">
     <palette>
      <active>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>0</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>0</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>120</red>
          <green>120</green>
          <blue>120</blue>
         </color>
        </brush>
       </colorrole>
      </disabled>
     </palette>
    </property>
    <property name="font">
     <font>
      <pointsize>120</pointsize>
     </font>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="cmdDebug">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>600</y>
      <width>100</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Debug</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="cmdFind">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>81</width>
      <height>81</height>
     </rect>
    </property>
    <property name="text">
     <string>Find File</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblExecutionTime">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>130</y>
      <width>120</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Execution Time</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblTime">
    <property name="geometry">
     <rect>
      <x>1130</x>
      <y>130</y>
      <width>171</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="cmdRelease">
    <property name="geometry">
     <rect>
      <x>1280</x>
      <y>650</y>
      <width>141</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Release</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="txtSerial">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>30</y>
      <width>331</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="lblSerial">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>0</y>
      <width>91</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Read serial</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cmdClear">
    <property name="geometry">
     <rect>
      <x>1240</x>
      <y>60</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear</string>
    </property>
   </widget>
   <zorder>lblPicture</zorder>
   <zorder>grpConfigs</zorder>
   <zorder>cmdOpen</zorder>
   <zorder>txtPathFile</zorder>
   <zorder>cmdStop</zorder>
   <zorder>cmdRun</zorder>
   <zorder>lstErrors</zorder>
   <zorder>lblErrors</zorder>
   <zorder>cmdLights</zorder>
   <zorder>cmdConfigs</zorder>
   <zorder>lblRows</zorder>
   <zorder>lblColumns</zorder>
   <zorder>lblNumberRows</zorder>
   <zorder>lblNumberColumns</zorder>
   <zorder>lblStatus</zorder>
   <zorder>cmdDebug</zorder>
   <zorder>cmdFind</zorder>
   <zorder>lblExecutionTime</zorder>
   <zorder>lblTime</zorder>
   <zorder>cmdRelease</zorder>
   <zorder>txtSerial</zorder>
   <zorder>lblSerial</zorder>
   <zorder>cmdClear</zorder>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>cmdFind</tabstop>
  <tabstop>txtPathFile</tabstop>
  <tabstop>cmdOpen</tabstop>
  <tabstop>cmdRun</tabstop>
  <tabstop>txtSerial</tabstop>
  <tabstop>lstErrors</tabstop>
  <tabstop>cmdConfigs</tabstop>
  <tabstop>cmdLights</tabstop>
  <tabstop>cmdStop</tabstop>
  <tabstop>txtLightUp</tabstop>
  <tabstop>txtLightDown</tabstop>
  <tabstop>txtLightLeft</tabstop>
  <tabstop>txtLightRight</tabstop>
  <tabstop>txtSensorUnit</tabstop>
  <tabstop>txtSolenoid</tabstop>
  <tabstop>txtButton</tabstop>
  <tabstop>cmdRelease</tabstop>
  <tabstop>cmdDebug</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>cmdOpen</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>OpenFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>696</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>84</x>
     <y>114</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdRun</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Run()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>939</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>1074</x>
     <y>89</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdStop</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>960</x>
     <y>173</y>
    </hint>
    <hint type="destinationlabel">
     <x>1102</x>
     <y>138</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdConfigs</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Configs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>945</x>
     <y>590</y>
    </hint>
    <hint type="destinationlabel">
     <x>1245</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdLights</sender>
   <signal>clicked(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>Lights(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1270</x>
     <y>591</y>
    </hint>
    <hint type="destinationlabel">
     <x>1279</x>
     <y>337</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdDebug</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Debug()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1059</x>
     <y>611</y>
    </hint>
    <hint type="destinationlabel">
     <x>1023</x>
     <y>342</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdFind</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Find()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>49</x>
     <y>35</y>
    </hint>
    <hint type="destinationlabel">
     <x>994</x>
     <y>254</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdRelease</sender>
   <signal>clicked(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>Release(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1367</x>
     <y>659</y>
    </hint>
    <hint type="destinationlabel">
     <x>1362</x>
     <y>744</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtSerial</sender>
   <signal>returnPressed()</signal>
   <receiver>MainWindow</receiver>
   <slot>SendSerial()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1130</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>1330</x>
     <y>782</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdClear</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1284</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>1404</x>
     <y>724</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>_Update()</signal>
  <slot>OpenFile()</slot>
  <slot>Run()</slot>
  <slot>Stop()</slot>
  <slot>Configs()</slot>
  <slot>Release(bool)</slot>
  <slot>Debug()</slot>
  <slot>Refresh()</slot>
  <slot>Find()</slot>
  <slot>Lights(bool)</slot>
  <slot>SendSerial()</slot>
  <slot>Clear()</slot>
 </slots>
</ui>
